### DISCUSSION TITLE
Add option to show changelogs for firmware updates

### DISCUSSION THREAD

**Discussion by LinuxOnTheDesktop** [fwupd/fwupd#3234] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I'm frustrated when I see something like the following and have very little idea of what the update will do.

```
$ fwupdmgr get-updates
• Thunderbolt Controller has no available firmware updates
• LENSE20512GMSP34MEAT2TA has the latest available firmware version
• System Firmware has the latest available firmware version
• UEFI Device Firmware has the latest available firmware version
• UEFI Device Firmware has the latest available firmware version
• Unifying Receiver has the latest available firmware version
20KH001UKP
│
└─UEFI Device Firmware:
  │   Device ID:           9e329270a7a68d289c82fe77d32d02208ddf0890
  │   Current version:     0.71.12
  │   Minimum Version:     0.71.12
  │   Vendor:              DMI:LENOVO
  │   GUID:                cea87551-1701-43fb-afbc-6e8ce9728345
  │   Device Flags:        • Internal device
  │                        • Updatable
  │                        • Requires AC power
  │                        • Supported on remote server
  │                        • Needs a reboot after installation
  │                        • Device is usable for the duration of the update
  │ 
  └─ThinkPad X1 Carbon 6th System Update:
        New version:       0.73.20
        Remote ID:         lvfs
        Summary:           Lenovo ThinkPad X1 Carbon 6th STM TPM Firmware
        License:           Proprietary
        Size:              439.6 kB
        Vendor:            Lenovo Ltd.
        Flags:             is-upgrade
        Description:       Lenovo ThinkPad X1 Carbon 6th STM TPM Firmware Version 73.20
        
         • Do NOT turn off your computer or remove the AC adaptor while update is in progress
```

After all, don't at least some vendors provide changelogs?

**Describe the solution you'd like**

`$ fwupdmgr get-updates --changelogs`

```
$ fwupdmgr get-updates
[. . . .]
• Use the --changelogs flag to see any changelogs
• Do NOT turn off your computer or remove the AC adaptor while update is in progress
```

**Describe alternatives you've considered**

**Additional context**

**Comment by hughsie** [comment]
> After all, don't at least some vendors provide changelogs?

Lenovo was supposed to enter a changelog, but the only information they entered was `Lenovo ThinkPad X1 Carbon 6th STM TPM Firmware Version 73.20`

**Comment by LinuxOnTheDesktop** [comment, marked as answer]
Hughsie, do you mean that fwupd does give changelogs if any changelog is given to fwupd? Presuming yes: that state of affairs is not made clear by fwupd.

**Reply by hughsie** [reply]
Yes, the changelog is called an "Update Description" in fwupd parlance.

**Comment by LinuxOnTheDesktop** [comment]
I see. Thank. I propose the following change to the output of `get-updates`. Replace 'Update description' with something like 'Update description (including any vendor-supplied changelog)'? That way the user, confronted with a very short description that lacks a changelog, will not think that there is a changelog which fwupd is not displaying.
